src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -802,93 +712,125 @@ private void VerifyCallAllowed(bool synchronousCall)
         }
 
         /// <summary>
-        /// Catch any exception thrown by the action passed in; in the exception case move the writer into
-        /// state ExceptionThrown and then rethrow the exception.
+        /// Verifies that calling CreateOperationRequestMessage is valid.
         /// </summary>
-        /// <param name="action">The action to execute.</param>
-        private void InterceptException(Action action)
+        /// <param name="synchronousCall">true if the call is to be synchronous; false otherwise.</param>
+        /// <param name="method">The Http method to be used for this request operation.</param>
+        /// <param name="uri">The Uri to be used for this request operation.</param>
+        /// <param name="contentId">The Content-ID value to write in ChangeSet head.</param>
+        private void VerifyCanCreateOperationRequestMessage(bool synchronousCall, string method, Uri uri,
+            string contentId)
         {
-            try
+            this.CanCreateOperationRequestMessageVerifierCommon(synchronousCall, method, uri, contentId);

```

Comment:
>CanCreateOperationRequestMessageVerifierCommon [](start = 17, length = 46)

The method "CanCreateOperationRequestMessageVerifierCommon" is only called from here. Any reason to make it a separate method, rather than just put the logic from that method here?  If we do keep it as a separate method, can we simplify the name to "CanCreateOperationRequestMessage"?  The "VerifierCommon" is no longer meaningful since it's only implemented in/called from the base class so it's by definition "common".

Reply:
Moved to inline logics.

---
In reply to: [143602084](https://github.com/OData/odata.net/pull/960#discussion_r143602084) [](ancestors = 143602084)